services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -0,0 +1,180 @@
+        }
+
+        when: "Execute port UP on the src switch"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Port is really UP"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LIVE"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+
+        and: "Port history is updated on the src switch"
+        Long timestampAfterUp = System.currentTimeSeconds()
+        with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterUp)) {
+            it.size() == 2
+            checkHistoryPortUpAction(it[0], isl.srcSwitch.dpId, isl.srcPort)
+        }
+
+        and: "Port history is not empty when link is direct"
+        northboundV2.getPortHistory(isl.dstSwitch.dpId, isl.dstPort, timestampBefore, timestampAfterUp).size() ==
+                historySizeOnDstSw
+
+        and: "Port history is also available using default timeline"
+        northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort).size() >= 2
+
+        where:
+        [islDescription, historySizeOnDstSw, isl] << [
+                ["direct", 2, getTopology().islsForActiveSwitches.find { !it.aswitch }],
+                ["a-switch", 0, getTopology().islsForActiveSwitches.find {
+                    it.aswitch?.inPort && it.aswitch?.outPort
+                }]
+        ]
+    }
+
+    @Tags(SMOKE)
+    def "Port history should not be returned in case timeline is incorrect (timeBefore > timeAfter)"() {
+        given: "A direct link"
+        def timestampBefore = System.currentTimeSeconds()
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+
+        when: "Execute port DOWN on the src switch"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LINK_DOWN"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "Execute port UP on the src switch"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LIVE"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Port history is created"
+        Long timestampAfter = System.currentTimeSeconds()
+        def portHistory = northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfter)
+        assert portHistory.size() == 2
+
+        when: "Get port history for incorrect timeline"
+        def portH = northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampAfter, timestampBefore)
+
+        then: "Port history is NOT returned"
+        portH.isEmpty()
```

Comment:
shouldn't we throw in case when 'from' is later than 'to'?

Reply:
no, we just try to get information from DB.(db returns nothing)
and actually it works as flow history